### PR TITLE
[Setting] Rename Data

### PR DESCRIPTION
**Pull request by lamriaimen**
According to the guidelines

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Sofa/GUI/Qt/src/sofa/gui/qt/GUI.ui [hunk 1]
```diff
@@ -517,7 +517,7 @@ State 2: dirty, in that state the button reflect the fact that the scene graph v
             <item>
              <widget class="QCheckBox" name="dumpStateCheckBox">
               <property name="toolTip">
-               <string comment="This is used to plut curves fter a simulation">Record the state at each time step in file &quot;dumpState.data&quot;</string>
+               <string comment="This is used to plut curves fter a simulation">Record the state at each time step in file &quot;d_dumpState.data&quot;</string>
```
Comment (alxbilger) [inline Sofa/GUI/Qt/src/sofa/gui/qt/GUI.ui]: This change is not appropriate. You must revert the change

File: Sofa/framework/Simulation/Core/src/sofa/simulation/Simulation.cpp [hunk 2]
```diff
@@ -412,7 +412,7 @@ void exportGraph(Node* root, const char* filename)
 
 void dumpState(Node* root, std::ofstream& out)
 {
-    SCOPED_TIMER_VARNAME(dumpStateTimer, "dumpState");
+    SCOPED_TIMER_VARNAME(dumpStateTimer, "d_dumpState");
```
Comment (alxbilger) [inline Sofa/framework/Simulation/Core/src/sofa/simulation/Simulation.cpp]:
```suggestion
    SCOPED_TIMER_VARNAME(dumpStateTimer, "dumpState");
```

File: Sofa/framework/Simulation/Core/src/sofa/simulation/config.h.in [hunk 3]
```diff
@@ -165,7 +165,7 @@ SOFA_ATTRIBUTE_DISABLED( \
 #else
 #define SOFA_ATTRIBUTE_DISABLED_SIMULATION_DUMPSTATE() \
 SOFA_ATTRIBUTE_DISABLED( \
-"v23.12", "v24.06", "Use sofa::simulation::node::dumpState")
+"v23.12", "v24.06", "Use sofa::simulation::node::d_dumpState")
```
Comment (alxbilger) [inline Sofa/framework/Simulation/Core/src/sofa/simulation/config.h.in]:
```suggestion
"v23.12", "v24.06", "Use sofa::simulation::node::dumpState")
```

File: Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp [hunk 4]
```diff
@@ -2069,7 +2069,7 @@ void RealGUI::dumpState ( bool value )
     m_dumpState = value;
     if ( m_dumpState )
     {
-        m_dumpStateStream = new std::ofstream ( "dumpState.data" );
+        m_dumpStateStream = new std::ofstream ( "d_dumpState.data" );
```
Comment (alxbilger) [inline Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp]:
```suggestion
        m_dumpStateStream = new std::ofstream ( "dumpState.data" );
```

File: Sofa/Component/Setting/src/sofa/component/setting/StatsSetting.cpp [hunk 5]
```diff
@@ -33,13 +33,17 @@ int StatsSettingClass = core::RegisterObject("Stats settings")
         ;
 
 StatsSetting::StatsSetting():
-    dumpState(initData(&dumpState,false,"dumpState", "Dump state vectors at each time step of the simulation"))
-    , logTime(initData(&logTime, false, "logTime", "Output in the console an average of the time spent during different stages of the simulation"))
-    , exportState(initData(&exportState, false, "exportState", "Create GNUPLOT files with the positions, velocities and forces of all the simulated objects of the scene"))
+        d_dumpState(initData(&d_dumpState, false, "d_dumpState", "Dump state vectors at each time step of the simulation"))
```
Comment (alxbilger) [inline Sofa/Component/Setting/src/sofa/component/setting/StatsSetting.cpp]:
```suggestion
        d_dumpState(initData(&d_dumpState, false, "dumpState", "Dump state vectors at each time step of the simulation"))
```

File: Sofa/Component/Setting/src/sofa/component/setting/StatsSetting.cpp [hunk 6]
```diff
@@ -33,13 +33,17 @@ int StatsSettingClass = core::RegisterObject("Stats settings")
...
+    , d_logTime(initData(&d_logTime, false, "logTime", "Output in the console an average of the time spent during different stages of the simulation"))
+    , d_exportState(initData(&d_exportState, false, "d_exportState", "Create GNUPLOT files with the positions, velocities and forces of all the simulated objects of the scene"))
```
Comment (alxbilger) [inline Sofa/Component/Setting/src/sofa/component/setting/StatsSetting.cpp]:
```suggestion
    , d_exportState(initData(&d_exportState, false, "exportState", "Create GNUPLOT files with the positions, velocities and forces of all the simulated objects of the scene"))
```

**Review by alxbilger — CHANGES_REQUESTED** [review]

File: Sofa/Component/Setting/src/sofa/component/setting/SofaDefaultPathSetting.cpp [hunk 7]
```diff
@@ -33,8 +33,8 @@ int SofaDefaultPathSettingClass = core::RegisterObject("Default Paths for Sofa A
         ;
 
 SofaDefaultPathSetting::SofaDefaultPathSetting():
-     gnuplotPath(initData(&gnuplotPath, "gnuplotPath", "Path where will be saved the gnuplot files"))
+        d_gnuplotPath(initData(&d_gnuplotPath, "d_gnuplotPath", "Path where will be saved the gnuplot files"))
```
Comment (bakpaul) [inline Sofa/Component/Setting/src/sofa/component/setting/SofaDefaultPathSetting.cpp]:
```suggestion
        d_gnuplotPath(initData(&d_gnuplotPath, "gnuplotPath", "Path where will be saved the gnuplot files"))
```

**Review by alxbilger — APPROVED** [review]

**Review by fredroy — APPROVED** [review]

**Comment by lamriaimen** [comment]
[ci-build][with-all-tests]

**Review by fredroy — APPROVED** [review]

**Pull request merged**